docs/docs/spec.md

```diff
@@ -467,7 +478,11 @@ condition is met, skip the remaining conditions.
     > This special case handles the fact that the Java compiler automatically
     > generates an implementation of `equals` in `Record` but does not include a
     > `@Nullable` annotation on its parameter, even when the class is
-    > `@NullMarked`.
+    > `@NullMarked`. Note that this special handling is not necessary for the
+    > return type of `String toString()`.
+
+    TODO: should this only apply if the method isn't manually declared in the record?
+    As written now, it would also apply for a manually implemented `equals`, which I would find inconsistent.
```

Comment:
We don't _want_ it to apply for a manually implemented `equals`, but there's no way to tell that apart from an automatically generated one in the class file :(

There's some discussion scattered throughout https://github.com/jspecify/jspecify/issues/301 and https://github.com/jspecify/jspecify/pull/645.

What we have at the moment is the best we've been able to come up with, with the combination of this and https://jspecify.dev/docs/spec/#expected-annotations-on-record-classes-equals-methods to make clear that people _should_ annotate any `equals` method that they hand-write but that we essentially "accidentally" accommodate them if they don't.

Reply:
Thanks for these links! I've adapted the text and included a forward reference. (You'll need to re-format again, sorry about that.)